modules/apps/frontend-js/frontend-js-collapse-support-web/src/main/resources/META-INF/resources/CollapseProvider.ts

```diff
@@ -117,6 +117,24 @@ class CollapseProvider {
 			trigger = this._getTrigger(panel);
 		}
 
+		// If the initial trigger isn't found, assume it could be a clay:panel
+		// taglib and attempt to expand it.
+
+		if (!trigger) {
+			const clayPanelTrigger = panel
```

Comment:
Why not using this selector in the `_getTrigger` method ([#L220](https://github.com/liferay-frontend/liferay-portal/pull/3705/files#diff-292c919c050e181ed724f4e42a067c3d476204bfc8a9836a507914d0e9c2b7e8R220)) and return the control to this provider?

Reply:
I needed a way of skipping the [rest of the `show` function](https://github.com/liferay-frontend/liferay-portal/blob/8cde2fe141f137c63adad919ec61858e3df09f4f/modules/apps/frontend-js/frontend-js-collapse-support-web/src/main/resources/META-INF/resources/CollapseProvider.ts#L167-L204) since it manipulates the DOM which messes up the React clay panel since the states aren't updated.

Reply:
Clear now. Thank you for the explanation